File: Seminar/src/main/java/org/sopt/Seminar/global/common/exception/ErrorType.java
package org.sopt.Seminar.global.common.exception;

public interface ErrorType {
    int getCode();

    enum BadRequest implements ErrorType {
        ;

        private final int errorCode;

        BadRequest(int errorCode) {
            this.errorCode = errorCode;
        }

        @Override
        public int getCode() {
            return this.errorCode;
        }
    }

    enum NotFound implements ErrorType {
        POST_NOT_FOUND(4001),
        USER_NOT_FOUND(4002)
        ;

        private final int errorCode;

        NotFound(int errorCode) {
            this.errorCode = errorCode;
        }

        @Override
        public int getCode() {
            return this.errorCode;
        }
    }

    enum ServerError implements ErrorType {
        SERVER_ERROR_DEFAULT(5000)
        ;

        private final int errorCode;

        ServerError(int errorCode) {
            this.errorCode = errorCode;
        }


        @Override
        public int getCode() {
            return this.errorCode;
        }
    }
}

File: kopring/week2/kotlin-semina/src/main/kotlin/org/sopt/kotlinsemina/global/common/exception/ErrorType.kt
package org.sopt.kotlinsemina.global.common.exception

interface ErrorType {
    val errorCode: Int

    enum class BadRequest(override val errorCode: Int): ErrorType {
        ;
    }

    enum class NotFound(override val errorCode: Int): ErrorType {
        POST_NOT_FOUND(4001),
        MEMBER_NOT_FOUND(4002),
        CATEGORY_NOT_FOUND(4003)
    }

    enum class ServerError(override val errorCode: Int): ErrorType {
        SERVER_ERROR_DEFAULT(5000)
    }
}